### PR TITLE
KAFKA-16754: Removing partitions from release API (KIP-932)

### PR DESCRIPTION
**Pull request by apoorvmittal10**
The release API exposed Partitions which should be an internal implementation detail for `releaseAcquiredRecords` API. Also lessen the scope for cached topic partitons method as it's not needed.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by apoorvmittal10** [comment]
The context: https://github.com/apache/kafka/pull/16456/files#r1662556294

File: core/src/test/java/kafka/server/share/SharePartitionManagerTest.java [hunk 1]
```diff
@@ -1246,26 +1250,40 @@ public void testCloseSharePartitionManager() throws Exception {
     @Test
     public void testReleaseAcquiredRecordsSuccess() {
         String groupId = "grp";
-        String memberId = Uuid.randomUuid().toString();
+        Uuid memberId = Uuid.randomUuid();
```
Comment (adixitconfluent) [inline core/src/test/java/kafka/server/share/SharePartitionManagerTest.java:1253]: Any particular reason toString method is removed here and then used at multiple places within the test wherever memberId is being used?

File: server/src/main/java/org/apache/kafka/server/share/ShareSessionCache.java [hunk 2]
```diff
@@ -41,10 +41,10 @@ public class ShareSessionCache {
...
     // A map of session key to ShareSession.
-    private Map<ShareSessionKey, ShareSession> sessions = new HashMap<>();
+    private final Map<ShareSessionKey, ShareSession> sessions = new HashMap<>();
 
     // Maps last used times to sessions.
-    private TreeMap<LastUsedKey, ShareSession> lastUsed = new TreeMap<>();
+    private final TreeMap<LastUsedKey, ShareSession> lastUsed = new TreeMap<>();
 
     // Visible for testing
     public synchronized TreeMap<LastUsedKey, ShareSession> lastUsed() {
```
Comment (AndrewJSchofield) [inline server/src/main/java/org/apache/kafka/server/share/ShareSessionCache.java]: This seems a bit odd. This method exposes the TreeMap for testing, which breaks encapsulation and surely risks synchronization problems if used for anything other than testing. So, is there any point in the method being declared synchronized? Accessing the TreeMap in the caller is not going to be synchronized.

**Review by AndrewJSchofield — CHANGES_REQUESTED** [review]

File: core/src/test/java/kafka/server/share/SharePartitionManagerTest.java (hunk 1 above)
Comment (apoorvmittal10) [reply core/src/test/java/kafka/server/share/SharePartitionManagerTest.java:1253]: Yeah, a conversion from string to Uuid was required later so don't see any point in converting to string first then back to Uuid from string.

File: server/src/main/java/org/apache/kafka/server/share/ShareSessionCache.java (hunk 2 above)
Comment (apoorvmittal10) [reply server/src/main/java/org/apache/kafka/server/share/ShareSessionCache.java]: Yeah, that's true this is bit odd and I have removed the method from the class itself. I don't see any usage of this method yet and if required by tests in future then we should send a copy of tree map instead the map itself.

**Review by adixitconfluent — APPROVED** [review]
Thanks for the PR. LGTM.

**Comment by apoorvmittal10** [comment]
@AndrewJSchofield @omkreddy The build passed with unrelated tests failure.

**Review by AndrewJSchofield — APPROVED** [review]

**Review by omkreddy — APPROVED** [review]
@apoorvmittal10 Thanks for the PR. LGTM

**Pull request merged**